server/src/main/java/org/elasticsearch/search/internal/ShardSearchLocalRequest.java

```diff
@@ -73,6 +73,7 @@
     private SearchSourceBuilder source;
     private Boolean requestCache;
     private long nowInMillis;
+    private Boolean allowPartialSearchResults;
```

Comment:
I wonder if we could resolve the value to true or false earlier (by consulting the cluster setting), for example could we maybe resolve this value in the TransportSearch action so the value is resolved on the coordinating node and consistent for all shard requests? Then this could be a `boolean`?

Reply:
I have code in the TransportSearchAction.executeSearch that resolves the setting but not all tests transits this section of code.  Consequently I was seeing NPEs from some tests when an unresolved `null` SearchRequest Boolean was being assigned to a `boolean` as a deeper part of query execution. I changed the failing tests to explicitly set SearchRequest.allowPartialSearchResults but it felt like a hack addressing the symptoms rather than the cause. Perhaps there should be a more explicit query preparation stage where user-supplied requests are augmented by any cluster-level options e.g. SearchRequest.fillInMissingParamsWithDefaults(Settings).